STACpopulator/implementations/Ouranos_CMIP6-CORDEX/add_CORDEX6.py

```diff
@@ -0,0 +1,75 @@
+import argparse
+import json
+import logging
+import os
+import sys
+from typing import Any, MutableMapping, Optional, Union
+
+from pystac import STACValidationError
+from pystac.extensions.datacube import DatacubeExtension
+from requests.sessions import Session
+
+from STACpopulator.requests import add_request_options, apply_request_options
+from STACpopulator.extensions.cmip6 import CMIP6Helper, CMIP6Properties
+from STACpopulator.extensions.datacube import DataCubeHelper
+from STACpopulator.extensions.thredds import THREDDSExtension, THREDDSHelper
+from STACpopulator.input import ErrorLoader, GenericLoader, THREDDSLoader
+from STACpopulator.models import GeoJSONPolygon
+from STACpopulator.populator_base import STACpopulatorBase
+
+LOGGER = logging.getLogger(__name__)
+import argparse
+from typing import Any
+from STACpopulator.populator_base import STACpopulatorBase
+from STACpopulator.extensions.cordex6 import Cordex6DataModel
+
+
+class CORDEX_STAC_Populator(STACpopulatorBase):
+    data_model = Cordex6DataModel
+    item_geometry_model = None  # Unnecessary, but kept for consistency
+
+    def create_stac_item(self, item_name: str, item_data: dict[str, Any]) -> dict[str, Any]:
+        dm = self.data_model.from_data(item_data)
+        return dm.stac_item()
```

Comment:
Note that that `THREDDSCatalogDataModel` automatically applies the `datacube` and `thredds` extension.

Reply:
One issue I see with this is that the extension helpers have different `__init__` requirements. So either the helpers know how to parse the input data, or the object instantiating them provides that logic.

Reply:
We could have an in-between solution. 
The `item_helpers` list could define instances rather than type references: 
```python
item_helpers = [
  HelperWithoutArg(),
  THREDDSHelper(["<url>"]),
]
```
Anything that can be supplied at init would be created right away, and the STAC `item` objects would be obtained during the `apply(item)` call. 

I don't think there are any cases where the helpers would be missing references limiting this approach, but to investigate...

Reply:
Not sure I follow. You need the data to create instances of the helpers.

Reply:
What we could do is something like this:
```python
    @classmethod
    def from_data(cls, data):
        """Instantiate class from data provided by THREDDS Loader.
        """
        # This is where we match the Loader's output to the STAC item and extensions inputs. If we had multiple
        # loaders, that's probably the only thing that would be different between them.
        return cls(data=data,
                   start_datetime=data["groups"]["CFMetadata"]["attributes"]["time_coverage_start"],
                   end_datetime=data["groups"]["CFMetadata"]["attributes"]["time_coverage_end"],
                   geometry=ncattrs_to_geometry(data),
                   bbox=ncattrs_to_bbox(data),
                   properties=data["attributes"],
                   )

    @model_validator(mode="before")
    @classmethod
    def datacube_helper(cls, data):
        """Validate the DataCubeHelper."""
        data["datacube"] = DataCubeHelper(data['data'])
        return data

    @model_validator(mode="before")
    @classmethod
    def thredds_helper(cls, data):
        """Validate the DataCubeHelper."""
        data["thredds"] = THREDDSHelper(data['data']["access_urls"])
        return data
```